contracts/multisig-improved/src/common_types/signature.rs

```diff
@@ -0,0 +1,87 @@
+use multiversx_sc::api::{CryptoApi, CryptoApiImpl};
+
+use super::action::{Action, GroupId, Nonce};
+
+multiversx_sc::imports!();
+multiversx_sc::derive_imports!();
+
+/// Board member, internal user nonce, action type (SimpleAction or Batch), SignatureType (Ed25519, Secp256r1 or Secp256k1) and raw signature
+pub type SignatureMultiArg<M> =
```

Comment:
can't we do this as a struct? wouldn't it be easier to reason about it?

Reply:
Done. Only reason I usually avoid it is because people often complain about structs as input.